tests/lib/sprintf/src/main.c

```diff
@@ -791,7 +791,8 @@ ZTEST(sprintf, test_fprintf)
 	ret = fprintf(stdout, "");
 	zassert_equal(ret, 0, "fprintf failed!");
 
-#if !defined(CONFIG_PICOLIBC) && !defined(CONFIG_ARMCLANG_STD_LIBC)	/* this is UB */
+#if !defined(CONFIG_PICOLIBC) && !defined(CONFIG_ARMCLANG_STD_LIBC) \
+	&& !__STDC_HOSTED__	/* this is UB */
 	ret = fprintf(NULL, "%d", i);
 	zassert_equal(ret, EOF, "fprintf failed!");
 #endif
```

Comment:
I'd prefer removing all UB from the tests if that's acceptable. Otherwise, we'd have to write  a spec for the minimal C library which specified where it differed from standard C, which really seems unhelpful.

Reply:
done.